util/hot_backup_test.cc

```diff
@@ -0,0 +1,600 @@
+    ret_val=mkdir(m_DBName.c_str(), 0777);
+    ret_val=mkdir(options.tiered_fast_prefix.c_str(), 0777);
+    ASSERT_TRUE(0==ret_val);
+    ret_val=mkdir(options.tiered_slow_prefix.c_str(), 0777);
+    ASSERT_TRUE(0==ret_val);
+    MakeLevelDirectories(Env::Default(), options);
+
+    backup_options=options;
+    SetBackupPaths(backup_options, 0);
+
+    // this loop goes one higher than permitted retention
+    //  to validate deletion of oldest
+    for (loop=0; loop<=config::kNumBackups; ++loop)
+    {
+        // rotate directories
+        ret_flag=PrepareDirectories(options);
+        ASSERT_TRUE(ret_flag);
+
+        // these files are to "mark" the backups, not
+        //  pretending this is a true file link
+        // make a file in fast tier ... 10 + backup iteration
+        table_file=TableFileName(backup_options, 10+loop, 1);
+        ret_val=open(table_file.c_str(), O_CREAT | O_RDWR, S_IRUSR | S_IWUSR);
+        ASSERT_TRUE(-1!=ret_val);
+        close(ret_val);
+
+        // make a file in slow tier ... 20 + backup iteration
+        table_file=TableFileName(backup_options, 20+loop, 5);
+        ret_val=open(table_file.c_str(), O_CREAT | O_RDWR, S_IRUSR | S_IWUSR);
+        ASSERT_TRUE(-1!=ret_val);
+        close(ret_val);
+
+        // test files
+        for (inner_loop=config::kNumBackups+1; 0<=inner_loop; --inner_loop)
+        {
+            offset=(loop<config::kNumBackups) ? 0 : loop-config::kNumBackups +1;
+            should_find=(inner_loop<=loop) && (offset<=inner_loop);
+
+            inner_options=options;
+            SetBackupPaths(inner_options, inner_loop-offset);
+
+            file_num=loop - inner_loop + offset;
+
+            table_file=TableFileName(inner_options, 10+file_num, 1);
+            ret_val=access(table_file.c_str(), F_OK);
+            ASSERT_TRUE( (0==ret_val) == should_find);
+
+            table_file=TableFileName(inner_options, 20+file_num, 5);
+            ret_val=access(table_file.c_str(), F_OK);
+            ASSERT_TRUE( (0==ret_val) == should_find);
+        }   // for
+    }   // for
+
+    ClearAllBackups(options);
+
+    // clean up ... other unit test programs leave stuff, this fails
+    ret_val=rmdir(m_DBName.c_str());
+    //ASSERT_TRUE(0==ret_val);
+
+}   // DirectoryRotationTest
+
+
+/**
+ * Test that code has access permissions to copy
+ * live LOG file.
+ */
+TEST(HotBackupTester, LOGCopyTest)
+{
+    Options options;
+    Status s;
+    DB * db;
+    WriteOptions write_ops;
+    std::string backup_path, backup_LOG;
+    long log_position;
+    bool ret_flag;
+    struct stat file_stat;
+    int ret_val;
+
+    backup_path=BackupPath(m_DBName, 0);
+    backup_LOG=InfoLogFileName(backup_path);
+
+    // preclean
+    DestroyDB(backup_path, options);
+    rmdir(backup_path.c_str());
+    DestroyDB(m_DBName, options);
+    rmdir(m_DBName.c_str());
+
+    // need a live database for this test.
+    options.create_if_missing=true;
+    options.error_if_exists;
```

Comment:
Yes, on OS X. Setting up to build on Debian as I write this.

On Sun, Aug 14, 2016 at 4:22 PM, Matthew Von-Maszewski <
notifications@github.com> wrote:

> In util/hot_backup_test.cc
> <https://github.com/basho/leveldb/pull/202#discussion_r74708676>:
>
> > +    bool ret_flag;
> > +    struct stat file_stat;
> > +    int ret_val;
> > +
> > +    backup_path=BackupPath(m_DBName, 0);
> > +    backup_LOG=InfoLogFileName(backup_path);
> > +
> > +    // preclean
> > +    DestroyDB(backup_path, options);
> > +    rmdir(backup_path.c_str());
> > +    DestroyDB(m_DBName, options);
> > +    rmdir(m_DBName.c_str());
> > +
> > +    // need a live database for this test.
> > +    options.create_if_missing=true;
> > +    options.error_if_exists;
>
> Oops. Where are you building? OS X? Want to duplicate the warning.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/basho/leveldb/pull/202/files/447356c5a4fa8905b3f998db9f227fe649317158#r74708676>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AMnddBEiM1sBWwWokgZ4XPpGxaz2UsXIks5qf4clgaJpZM4JcAvk>
> .
>